File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/GameTimeoutAction.java
package org.emulinker.kaillera.controller.v086.action;

import com.google.common.flogger.FluentLogger;
import javax.inject.Inject;
import javax.inject.Singleton;
import org.emulinker.kaillera.controller.v086.V086Controller.V086ClientHandler;
import org.emulinker.kaillera.model.KailleraUser;
import org.emulinker.kaillera.model.event.*;

@Singleton
public class GameTimeoutAction implements V086GameEventHandler<GameTimeoutEvent> {
  private static final FluentLogger logger = FluentLogger.forEnclosingClass();

  private static final String DESC = "GameTimeoutAction";

  private int handledCount = 0;

  @Inject
  GameTimeoutAction() {}

  @Override
  public int getHandledEventCount() {
    return handledCount;
  }

  @Override
  public String toString() {
    return DESC;
  }

  @Override
  public void handleEvent(GameTimeoutEvent timeoutEvent, V086ClientHandler clientHandler) {
    handledCount++;

    KailleraUser player = timeoutEvent.getUser();
    KailleraUser user = clientHandler.getUser();

    if (player.equals(user)) {
      logger.atFine().log(
          user
              + " received timeout event "
              + timeoutEvent.getTimeoutNumber()
              + " for "
              + timeoutEvent.getGame()
              + ": resending messages...");
      clientHandler.resend(timeoutEvent.getTimeoutNumber());
    } else {
      logger.atFine().log(
          user
              + " received timeout event "
              + timeoutEvent.getTimeoutNumber()
              + " from "
              + player
              + " for "
              + timeoutEvent.getGame());
    }
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/GameTimeoutAction.kt
package org.emulinker.kaillera.controller.v086.action

import com.google.common.flogger.FluentLogger
import javax.inject.Inject
import javax.inject.Singleton
import org.emulinker.kaillera.controller.v086.V086ClientHandler
import org.emulinker.kaillera.model.event.GameTimeoutEvent

private val logger = FluentLogger.forEnclosingClass()

@Singleton
class GameTimeoutAction @Inject internal constructor() : V086GameEventHandler<GameTimeoutEvent> {
  override var handledEventCount = 0
    private set

  override fun toString() = DESC

  override fun handleEvent(event: GameTimeoutEvent, clientHandler: V086ClientHandler) {
    handledEventCount++
    val player = event.user
    val user = clientHandler.user
    if (player == user) {
      logger
          .atFine()
          .log(
              "$user received timeout event ${event.timeoutNumber} for ${event.game}: resending messages...")
      clientHandler.resend(event.timeoutNumber)
    } else {
      logger
          .atFine()
          .log(
              "${user.toString()} received timeout event ${event.timeoutNumber} from $player for ${event.game}")
    }
  }

  companion object {
    private const val DESC = "GameTimeoutAction"
  }
}
